tests/testthat/test-read.R

```diff
@@ -0,0 +1,30 @@
+test_that("can read file with no helpers", {
+  expect_equal(orderly_read_r("examples/implicit/orderly.R"), list())
+})
+
+
+test_that("can read file with helpers", {
+  dat <- orderly_read_r("examples/explicit/orderly.R")
+  expect_setequal(names(dat), c("resources", "artefacts"))
+  expect_equal(dat$resources, "data.csv")
+  expect_equal(dat$artefacts,
+               list(list(description = "A graph of things",
+                         files = "mygraph.png")))
+})
+
+
+test_that("Can read string literals from expressions", {
+  expect_equal(static_character_vector(quote("x")), "x")
+  expect_equal(static_character_vector(quote(c("x"))), "x")
+  expect_equal(static_character_vector(quote(c("x", "y"))), c("x", "y"))
+  expect_equal(static_character_vector(quote(c("x", c("y", "z")))),
+               c("x", "y", "z"))
+
+  expect_null(static_character_vector(quote(a)))
```

Comment:
So we can't pass a variable into `orderly_resource` or `orderly_artefact`? That makes sense, but I wonder if need to improve the error messaging if you try and do that? Perhaps something for later, if you do e.g.

```
my_file <- "data.csv"
orderly3::orderly_resource(my_file)
```

when trying to run this with `orderly_run` it gives
```
>   id <- orderly_run("explicit", root = path, envir = env)
Error in static_character_vector(files) : object 'my_file' not found
```

Reply:
thanks, this is a bug, we should be able to do that. I'll have  look why that's not getting the right environment